test/framework/kubernetes.go

```diff
@@ -280,3 +284,69 @@ func checkPodContainerHasEnvVariableWithValue(pod *corev1.Pod, containerName, en
 	}
 	return false
 }
+
+// GetIngressURI returns the ingress URI
+func GetIngressURI(namespace, serviceName string) (string, error) {
+	ingress := &v1beta1.Ingress{}
+	if exists, err := kubernetes.ResourceC(kubeClient).FetchWithKey(types.NamespacedName{Name: serviceName, Namespace: namespace}, ingress); err != nil {
+		return "", err
+	} else if !exists {
+		return "", fmt.Errorf("Ingress %s does not exist in namespace %s", serviceName, namespace)
+	} else if len(ingress.Spec.Rules) == 0 {
+		return "", fmt.Errorf("Ingress %s does not have any rules", serviceName)
+	}
+
+	return fmt.Sprintf("http://%s:80", ingress.Spec.Rules[0].Host), nil
+}
+
+// ExposeServiceOnKubernetes adds ingress CR to expose a service
+func ExposeServiceOnKubernetes(namespace, serviceName string) error {
+	host := serviceName
+	if !config.IsLocalCluster() {
+		host += fmt.Sprintf(".%s.%s", namespace, config.GetDomainSuffix())
+	}
+
+	ingress := v1beta1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        serviceName,
+			Namespace:   namespace,
+			Annotations: map[string]string{"nginx.ingress.kubernetes.io/rewrite-target": "/"},
+		},
+		Spec: v1beta1.IngressSpec{
+			Rules: []v1beta1.IngressRule{
+				{
+					Host: host,
+					IngressRuleValue: v1beta1.IngressRuleValue{
+						HTTP: &v1beta1.HTTPIngressRuleValue{
+							Paths: []v1beta1.HTTPIngressPath{
+								{
+									Path: "/",
+									Backend: v1beta1.IngressBackend{
+										ServiceName: serviceName,
+										ServicePort: intstr.FromInt(8080),
```

Comment:
We should get existing port (retrieved from service?) instead of hardcoding as port will be configurable (https://github.com/kiegroup/kogito-cloud-operator/pull/378).

Reply:
Will do once this PR is merged.

Reply:
Raised https://issues.redhat.com/browse/KOGITO-2466